osbuild/util/parsing.py

```diff
@@ -72,16 +72,24 @@ def parse_input(url: ParseResult, args: Dict) -> os.PathLike:
     return root
 
 
-def parse_location(location: str, args: Dict) -> str:
+def parse_location_into_parts(location: str, args: Dict) -> Tuple[str, str]:
     """
-    Parses the location URL to derive the corresponding file path.
+    Parses the location URL to derive the corresponding root and url path.
 
     Parameters:
     - location (str): The location URL to be parsed.
     - args (Dict): A dictionary containing arguments including mounts and
     path information as passed by osbuild.api.arguments()
     """
 
+    if "://" not in location:
+        print("INFO: location has no scheme, assuming 'tree://'")
```

Comment:
Hm, given that this is called from `parse_location()` (which did not require the tree:// prefix) won't that produce a bunch of extra log nosie in an osbuild run? Or am I reading this wrong?

Reply:
It shouldn't. Before that function would raise an exception. But i can drop this `INFO...`, wdyt?

Reply:
> won't that produce a bunch of extra log nosie in an osbuild run?

some extra noise.. I don't think it will be that bad.

Reply:
Like Nikida said, so far the `parse_location()` function would raise an exception if you called it without a scheme, which means we shouldn't be getting into this scenario with any existing caller to the function.

Which makes me wonder why we need to support this now here.  If so far the calls to `parse_location()` required a scheme, and stages that took "bare" paths were responsible for pre-pending `tree://` before calling it (like `org.osbuild.mkdir`), then there's no real reason to print this info here, I think.  It'd be understandable if we wanted to change the behaviour of the function to make it a bit more strict but had callers that relied on the old functionality, but we're doing the opposite here.

I think there are two cleaner options here:
1. Require a scheme for all locations passed to `parse_location()` and `parse_location_into_parts()`, raising an exception when it's missing, or
2. Make it explicit and documented that paths without schemes are treated as `tree://`.

We do 2 now but print an INFO line about it and I think that's a bit weird.  If we really want that behaviour, let's just put it in the docstring and be silent about it.

Though, tbh, I think I prefer option 1.

Reply:
Then lets put it in the docstring that schemaless implies `tree://` and do it silently, I guess.